pkg/probe/option.go

```diff
@@ -27,6 +30,15 @@ type Options struct {
 }
 
 // IsValid returns true if some values are filled into the options.
-func (o *Options) IsValid() bool {
-	return o != nil && o.Path != "" && o.UpdateInterval > 0*time.Second
+func (o *Options) Validate() error {
```

Comment:
I think this is fine. Just one thought, however: It might be nice to return all validation errors in one go-round.

Reply:
Agree. Use `multierror.Append` to append errors